### PR TITLE
Add PS5 Package Management resources

### PR DESCRIPTION
**Pull request by trevorghess**
Work In Progress adding PS5 Package Management resources
-powershell_package
-powershell_package_provider
-powershell_package_source

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (smurawski) [inline .vscode/settings.json]: you should remove this or add it to the gitignore.

File: recipes/powershell5.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@
 case node['platform']
 when 'windows'
 
-  if ::Windows::VersionHelper.nt_version(node) >= 6.1
+  if ::Windows::VersionHelper.nt_version(node) >= 6.1 && ::Windows::VersionHelper.nt_version(node) < 10
```
Comment (smurawski) [inline recipes/powershell5.rb]: This shouldn't really matter - since the not_if should keep the resource from attempting to install, but either way is fine...

**Comment by smurawski** [comment]
Serverspec?  Really?  This is the powershell cookbook - write some Pester :)

File: .kitchen.cloud.yml [hunk 3]
```diff
@@ -36,7 +36,34 @@ platforms:
...
+      image_urn: MicrosoftWindowsServer:WindowsServer:2008-R2-SP1:latest
+      winrm_powershell_script: |-
+        winrm quickconfig -q
+        winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="512"}'
+        winrm set winrm/config '@{MaxTimeoutms="1800000"}'
+        winrm set winrm/config/service '@{AllowUnencrypted="true"}'
+        winrm set winrm/config/service/auth '@{Basic="true"}'
+        netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" profile=public protocol=tcp localport=5985 remoteip=localsubnet new remoteip=any
+    provisioner:
+      require_chef_omnibus: 12.5.1
```
Comment (smurawski) [inline .kitchen.cloud.yml]: Why did you change this to specifying 12.5.1 and not just letting it pick the latest 12 client?

Comment (trevorghess) [inline .kitchen.cloud.yml]: 12.6.0 has this [issue](https://github.com/chef/chef/issues/4354) which has been resolved but a new version of client hasn't been cut yet.
